docs/source/tutorials/profiling.ipynb

```diff
@@ -5,11 +5,375 @@
+    "When tracing your application, states trade off granularity for reliability. As such, they are also easier to use as they require a less granular and specific understanding of RP. To see the list of states, ordered in their guaranteed sequence, see RP's [state model](../internals.rst#state-model).\n",
+    "\n",
+    "## Record format\n",
+    "\n",
+    "Each `*.prof` file uses the following format:\n",
+    "\n",
+    "```\n",
+    "time,event,comp,thread,uid,state,msg\n",
+    "```\n",
+    "\n",
+    "Where:\n",
+    "\n",
+    "- `time` = (mandatory) timestamp expressed in Unix Time (epoch)\n",
+    "- `event` = (mandatory) the name of the timestamped event\n",
+    "- `comp` = (facultative) the name of the component that triggered the event\n",
+    "- `thread` = (mandatory) the ID of the thread in which the event was triggered with `comp`\n",
+    "- `uid` = (mandatory) the ID of the entity to which that event belongs (e.g., task or pilot)\n",
+    "- `state` = (facultative) the name given to the event when it is also a state, i.e., guaranteed to be triggered in every configuration of RP\n",
+    "- `msg` = (facultative) an arbitrary message associated to that event.\n",
+    "\n",
+    "For example, here is the record of the `launch_start` event for a task:\n",
+    "\n",
+    "```\n",
+    "1684787127.2228520,launch_start,,MainThread,task.000000,AGENT_EXECUTING,\n",
+    "```\n",
+    "\n",
+    "As you will see later in this tutorial, you can use `launch_start` to monitor when a task is/was executed. Here is another example, this time for the `schedule_ok` event triggered by RP's Agent Scheduler:\n",
+    "\n",
+    "```\n",
+    "1684787127.1614347,schedule_ok,agent_scheduling.0000,Thread-1 (_worker_thread),task.000000,,\n",
+    "```\n",
+    "\n",
+    "First note that this event is not a state, e.g., it would not fire in a configuration of RP where the scheduler is bypassed (a corner case as, for every practical purpose, this event will also fire!). As described in the [event model](../internals.rst#event-model), `schedule_ok` indicates the time at which the scheduler's `search for task resources succeeded`. Tracing this event for each task would give you the time series of when all your tasks were/are scheduled by RP for execution.\n",
+    "\n",
+    "<div class=\"alert alert-info\">\n",
+    "    \n",
+    "__Note:__ Order matters! States are guaranteed to be triggered always in the same order while only a subset of events have a partial order within each module. Consider that partial order when consulting [event model](../internals.rst#event-model).\n",
+    "\n",
+    "</div>\n",
+    "\n",
+    "\n",
+    "## RADICAL Analytics\n",
+    "\n",
+    "Parsing and filtering the `*.prof`files requires a solid understanding of: \n",
+    "\n",
+    "1. RP's event/state model in order to select only the information you actually need;\n",
+    "2. the format in which the information is encoded within in `*.prof` file;\n",
```

Comment:
``encoded within in `*.prof` file`` -> ``encoded within `*.prof` file`` (remove `in`)?